addons/analytic/tests/test_analytic_account.py

```diff
@@ -211,3 +212,16 @@ def test_all_account_count_with_subplans(self):
             with self.subTest(plan=plan.name, expected_count=expected_value):
                 with Form(plan) as plan_form:
                     self.assertEqual(plan_form.record.all_account_count, expected_value)
+
+    def test_analytic_account_branches(self):
+        """
+        Test that an analytic account defined in a parent company is accessible in its branches (children)
+        """
+        self.analytic_account_1.company_id = self.company_data
+        analytic_line = self.env['account.analytic.line'].create({
+            'name': 'company specific account',
+            'account_id': self.analytic_account_1.id,
+            'amount': 100,
+            'company_id': self.company_b_branch.id,
+        })
+        self.assertEqual(analytic_line.name, "company specific account")
```

Comment:
Not sure that the assertion makes sense :thinking: 
Just the fact that you can create should be enough

Reply:
two schools of thought, won't fight about it :D